quadriga/src/test/java/edu/asu/diging/quadriga/core/service/impl/CollectionManagerImplTest.java

```diff
@@ -44,11 +54,220 @@ public boolean matches(Collection arg0) {
                 return (arg0.getName().equals(name) && arg0.getDescription().equals(desc));
             }
         }))).thenReturn(savedCollection);
-        
+
         Collection collection = managerToTest.addCollection(name, desc);
         Assert.assertEquals(id, collection.getId());
         Assert.assertEquals(name, collection.getName());
         Assert.assertEquals(desc, collection.getDescription());
     }
 
+    @Test
+    public void test_findCollection_success() {
+        Collection collection = new Collection();
+        ObjectId objectId = new ObjectId();
+        collection.setId(objectId);
+        Mockito.when(collectionRepo.findById(objectId)).thenReturn(Optional.of(collection));
+        Collection foundCollection = managerToTest.findCollection(objectId.toString());
+        Assert.assertEquals(collection.getId().toString(), foundCollection.getId().toString());
+    }
+
+    @Test
+    public void test_editCollection_success() throws Exception {
+
+        String name = COLLECTION_NAME;
+        String desc = COLLECTION_DESC;
+        Collection existingCollection = new Collection();
+        ObjectId id = new ObjectId();
+        existingCollection.setId(id);
+        existingCollection.setName(name);
+        existingCollection.setDescription(desc);
+
+        Mockito.when(collectionRepo.findById(id)).thenReturn(Optional.of(existingCollection));
+
+        String editedName = EDITED_NAME;
+        String editedDescription = EDITED_DESC;
+
+        Collection editedCollection = new Collection();
+        editedCollection.setId(id);
+        editedCollection.setName(editedName);
+        editedCollection.setDescription(editedDescription);
+
+        Mockito.when(collectionRepo.save(Mockito.argThat(new ArgumentMatcher<Collection>() {
+
+            @Override
+            public boolean matches(Collection arg0) {
+                return (arg0.getName().equals(editedName) && arg0.getDescription().equals(editedDescription));
+            }
+
+        }))).thenReturn(editedCollection);
+
+        Collection updatedCollection = managerToTest.editCollection(id.toString(), editedName, editedDescription);
+        Assert.assertEquals(id.toString(), updatedCollection.getId().toString());
+        Assert.assertEquals(editedName, updatedCollection.getName());
+        Assert.assertEquals(editedDescription, updatedCollection.getDescription());
+
+    }
+
+    @Test
+    public void test_editCollection_nullDescription() throws Exception {
+
+        String name = COLLECTION_NAME;
+        String desc = COLLECTION_DESC;
```

Comment:
both of these variables seem to be used just once, so there is not really a point in creating them. just use the constant instead.